s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/config/S3SourceConfig.java

```diff
@@ -16,32 +16,356 @@
 
 package io.aiven.kafka.connect.s3.source.config;
 
-import java.util.Collections;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
+import org.apache.kafka.common.config.AbstractConfig;
 import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
 
+import io.aiven.kafka.connect.common.config.validators.NonEmptyPassword;
+import io.aiven.kafka.connect.common.config.validators.UrlValidator;
+import io.aiven.kafka.connect.s3.source.output.OutputFormat;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.RegionUtils;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.internal.BucketNameUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 @SuppressWarnings({ "PMD.TooManyMethods", "PMD.GodClass", "PMD.ExcessiveImports" })
-final public class S3SourceConfig {
+final public class S3SourceConfig extends AbstractConfig {
```

Comment:
There is an AivenCommonsConfig(which extends AbstractConfig) that comes with some kafka back offs and other defaults, can this be re-used here?

Reply:
That is mainly for sink configs. We will have to rename that to AivenCommonsSinkConfig, and introduce later a new one for source